### DISCUSSION TITLE
Is it possible to use Dialog with hook like useToast() ?

### DISCUSSION THREAD

**Discussion by akouyate** [chakra-ui/chakra-ui#3209] [General]
I would like to trigger Dialog modal from function calll. I have to use the same action flow in multiple corner of my app and I would like to avoid to implement dialog component at any time. That's could be better if I can display a dialog like that

``` JAVASCRIPT
dialog({
  title: "toto",
  message: "My message",
  onReject: () => {},
  onAccept: () => {}
}}
```

Someone already implement this feature. May be I should create my own hook with HOC.

**Comment by Jmg-21** [comment]
Also waiting for this feature, very handly <3

**Reply by vitalijalbu** [reply]
hi, you found a solution??

**Comment by dev-techmoe** [comment]
https://github.com/alexanderkhivrych/use-modal-hook seems a solution for this feature

Tip: don't forget wrap `Modal` component with `ChakraProvider`

**Comment by vitalijalbu** [comment]
> I would like to trigger Dialog modal from function calll. I have to use the same action flow in multiple corner of my app and I would like to avoid to implement dialog component at any time. That's could be better if I can display a dialog like that
> 
> ```js
> dialog({
>   title: "toto",
>   message: "My message",
>   onReject: () => {},
>   onAccept: () => {}
> }}
> ```
> 
> Someone already implement this feature. May be I should create my own hook with HOC.

Hi, did you found a solution?

**Comment by vitalijalbu** [comment]
Hi guys, for you:

[https://gist.github.com/vitalijalbu/1a62617e022d7b57b50a0dc300c90567](https://gist.github.com/vitalijalbu/1a62617e022d7b57b50a0dc300c90567)
